src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -65,12 +67,23 @@ public class QueryExecutor<TConnection> : IQueryExecutor
             IDictionary<string, DbConnectionParam> parameters,
             Func<DbDataReader, List<string>?, Task<TResult>>? dataReaderHandler,
             HttpContext? httpContext = null,
-            List<string>? args = null)
+            List<string>? args = null,
+            string dataSourceName = "")
         {
+            if (string.IsNullOrEmpty(dataSourceName))
```

Comment:
Will this condition and providing empty string as default be removed in the next PR i.e. all the interfaces will know the exact datasource to use?